src/components/directory-content.js

```diff
@@ -916,28 +1046,60 @@ const DirectoryContent = () => {
                 open={openScriptContingencyDialog}
                 onClose={handleCloseScriptContingency}
                 onError={handleError}
                 title={useIntl().formatMessage({ id: 'editContingencyList' })}
+                type={elementType.SCRIPT_CONTINGENCY_LIST}
+            />
+            <ScriptDialog
+                id={currentScriptId}
+                open={openScriptDialog}
+                onClose={handleCloseScriptDialog}
+                onError={handleError}
+                title={useIntl().formatMessage({ id: 'editFilterScript' })}
+                type={elementType.SCRIPT}
             />
-            <ContingencyReplaceWithScriptDialog
-                listId={activeElement ? activeElement.elementUuid : ''}
+            <ReplaceWithScriptDialog
+                id={activeElement ? activeElement.elementUuid : ''}
                 open={openFiltersContingencyReplaceWithScriptDialog}
                 onClose={handleCloseFiltersContingencyReplaceWithScript}
+                onClick={handleClickFiltersContingencyReplaceWithScript}
                 onError={handleError}
                 title={useIntl().formatMessage({ id: 'replaceList' })}
             />
-            <ContingencyCopyToScriptDialog
-                listId={activeElement ? activeElement.elementUuid : ''}
+            <CopyToScriptDialog
+                id={activeElement ? activeElement.elementUuid : ''}
                 open={openFiltersContingencyCopyToScriptDialog}
                 onClose={handleCloseFiltersContingencyCopyToScript}
-                onError={handleError}
+                onClick={handleClickContingencyCopyToScript}
                 currentName={activeElement ? activeElement.elementName : ''}
                 title={useIntl().formatMessage({ id: 'copyToScriptList' })}
             />
+            <ReplaceWithScriptDialog
+                id={activeElement ? activeElement.elementUuid : ''}
+                open={openFiltersReplaceWithScriptDialog}
+                onClose={handleCloseFiltersReplaceWithScript}
+                onClick={handleClickFiltersReplaceWithScript}
+                title={useIntl().formatMessage({ id: 'replaceList' })}
+            />
+            <CopyToScriptDialog
+                id={activeElement ? activeElement.elementUuid : ''}
+                open={openFiltersCopyToScriptDialog}
+                onClose={handleCloseFiltersCopyToScript}
+                onClick={handleClickFilterCopyToScript}
+                currentName={activeElement ? activeElement.elementName : ''}
+                title={useIntl().formatMessage({ id: 'copyToScriptList' })}
+            />
+            <GenericFilterDialog
+                id={currentFilterId}
+                open={openGenericFilterDialog}
+                onClose={handleCloseGenericFilterDialog}
+                onError={handleError}
+                title={useIntl().formatMessage({ id: 'editFilter' })}
+            />
```

Comment:
I'm not really fond of multiplying n dialogs for each element/action couple; a least not in an already big file.